tests/test_utils.py

```diff
@@ -111,3 +116,19 @@ def test_parse_header_links(value, expected):
 
     # Reset the logger so we don't have verbose output in all unit tests
     logging.getLogger("httpx").handlers = []
+
+
+def test_get_ssl_cert_file():
```

Comment:
This should also test if all three are set and if none are valid files, and verify the order tested.

Reply:
I'd probably tend to be okay either way on this sort of thing.
We can't ever test every possible permutation, and test code does have it's own cost.